components/fxa-client/ffi/src/lib.rs

```diff
@@ -414,15 +414,22 @@ pub extern "C" fn fxa_set_device_name(
 
 /// Fetch the devices (including the current one) in the current account.
 ///
+/// Devices might get cached in-memory and the caller might get served a cached version.
+/// To bypass this, the `ignore_cache` parameter can be set to `true`.
+///
 /// # Safety
 ///
 /// A destructor [fxa_bytebuffer_free] is provided for releasing the memory for this
 /// pointer type.
 #[no_mangle]
-pub extern "C" fn fxa_get_devices(handle: u64, error: &mut ExternError) -> ByteBuffer {
+pub extern "C" fn fxa_get_devices(
+    handle: u64,
+    ignore_cache: bool,
```

Comment:
For reasons that I won't pretend to understand in detail, we're not supposed to use `bool` at the FFI layer:

  https://github.com/mozilla/application-services/blob/master/docs/howtos/when-to-use-what-in-the-ffi.md#primitives

That `fxa_get_profile` currently does use it, is probably an oversight. Could you please adjust both `fxa_get_devices` and `fxa_get_profile` to use a `u8` here rather than `bool`, following the [example of the places FFI](https://github.com/mozilla/application-services/blob/5b130877dc5f5f8e4ebd8f0c1f4727d21a6c4e33/components/places/ffi/src/lib.rs#L266)? (Unfortunately this means converting it to a `u8` inside the [android](https://github.com/mozilla/application-services/blob/5b130877dc5f5f8e4ebd8f0c1f4727d21a6c4e33/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt#L318) and ios binding layers 😞)

Reply:
Thanks! I was somewhat unsure about the FFI layer changes I made so this comment and the link are very helpful.